Develop3

tigralbus opened the pull request.

Commit:
1 ver

Commit:
2 ver

Commit:
3 ver

Commit:
4 ver

Commit:
4 ver

Commit:
5 ver

Commit:
6 ver before api

Commit:
7 ver  api

Commit:
8 ver  api

Commit:
9 ver  api

Commit:
final ver

KseJha commented on constants.py:
```diff
@@ -0,0 +1,14 @@
+class Constants:
+    URL = 'https://stellarburgers.nomoreparties.site/'
```
Нужно исправить: нужно конкатенировать урлы, чтобы потом можно было быстро запускать тесты на разных окружениях. Пример:

URL = 'https://stellarburgers.nomoreparties.site/'
PERSONAL_ACCOUNT_URL = f'{URL}account/profile'

KseJha commented on pages/base_page.py:
```diff
@@ -0,0 +1,133 @@
+    def click_constructor_link(self):
+        self.click_locator(BasePageLocators.CONSTRUCTOR_LINK)
+
+    @allure.step('кликнуть на Ленту Заказов')
```
Нужно исправить: нужно внимательно посмотреть на все методы базового класса и оставить только методы для общей работы с элементами - ожидания, клики, получения текста и т.д. Подсказка - если метод импортирет локаторы, то скорее всего его нужно перенести в соответвующий page object класс.

tigralbus replied:
Привет, 
1) Я создала отдельный файл локаторов для хедеров - heders_page_locators
2) Все методы из base_page, которые используют локаторы других страниц и те, что сложнее взаимодействия с элементами, перенесла в новый вспомогательный класс HavigationHelper.
3) HavigationHelper вызывается в конструкторе класса каждой страницы.
4) В base_page_locators оставила функцию определения локатора, так как она нудна для 2 страниц.

Commit:
fix comment:
Нужно исправить: нужно конкатенировать урлы, чтобы потом можно было быстро запускать тесты на разных окружениях. Пример:

URL = 'https://stellarburgers.nomoreparties.site/'
PERSONAL_ACCOUNT_URL = f'{URL}account/profile'

Commit:
fix comment:
Нужно исправить: нужно внимательно посмотреть на все методы базового класса и оставить только методы для общей работы с элементами - ожидания, клики, получения текста и т.д. Подсказка - если метод импортирет локаторы, то скорее всего его нужно перенести в соответвующий page object класс.
